src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -294,7 +295,9 @@ def load_arguments(self, _):
         c.argument('include_parameter_default_value', action='store_true')
 
     with self.argument_context('group create') as c:
-        c.argument('rg_name', options_list=['--name', '--resource-group', '-n', '-g'], help='name of the new resource group', completer=None)
+        c.argument('rg_name', options_list=['--name', '--resource-group', '-n', '-g'],
+                   help='name of the new resource group', completer=None, lc_actions=[STORE], lc_scopes=[GLOBAL],
+                   lc_name='resource_group_name')
```

Comment:
Let's extract `lc_actions`, `lc_scopes` and `lc_name` to a `LocalContext` class, like `deprecate_info` https://github.com/Azure/azure-cli/blob/cef20494063f0bce9778220b6f870415d8914952/src/azure-cli/azure/cli/command_modules/redis/commands.py#L34

https://github.com/microsoft/knack/blob/5f0bcc2ae416c8f2834db71b49040976e1d7a29d/knack/commands.py#L377-L379

```py
    def deprecate(self, **kwargs):
        kwargs['object_type'] = 'command'
        return Deprecated(self.command_loader.cli_ctx, **kwargs)
```

Reply:
I used a `LocalContextAttribute` class and it is a little bit different with `Deprecated`. Please take a look at it.